apps/services/auth/delegation-api/src/app/clients/clients.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { Controller, Get, Query, UseGuards } from '@nestjs/common'
+import { ApiSecurity, ApiTags } from '@nestjs/swagger'
+
+import { ClientsService, ResourcesService } from '@island.is/auth-api-lib'
+import { IdsUserGuard, Scopes, ScopesGuard } from '@island.is/auth-nest-tools'
+import { AuthScope } from '@island.is/auth/scopes'
+import { Audit } from '@island.is/nest/audit'
+import { Documentation } from '@island.is/nest/swagger'
+
+import { ClientDto } from './client.dto'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(AuthScope.delegations)
+@ApiSecurity('ias', [AuthScope.delegations])
+@ApiTags('/clients')
+@Controller({
+  path: 'clients',
+  version: ['1'],
+})
+@Audit({ namespace: '@island.is/auth/delegation-api/clients' })
+export class ClientsController {
+  constructor(
+    private readonly clientsService: ClientsService,
+    private readonly resourcesService: ResourcesService,
+  ) {}
+
+  @Get()
+  @Documentation({
+    response: { status: 200, type: [ClientDto] },
+    request: {
+      query: {
+        lang: {
+          description: 'The language to return the client name in.',
+          required: false,
+          type: 'string',
+        },
+      },
+    },
+  })
+  @Audit<ClientDto[]>({
+    resources: (clients) => clients.map((client) => client.clientId),
+  })
+  async findAll(@Query('lang') lang?: string): Promise<ClientDto[]> {
+    const clients = await this.clientsService.findAllWithTranslation(lang)
+
+    if (!clients) {
+      return []
+    }
+
+    return Promise.all(
+      clients.map(async (client) => {
+        // Todo: This is a temporary solution until we have linked clients to domains
+        const clientDomain = client.clientId.split('/')[0]
+        const domain = await this.resourcesService.findDomainByPk(clientDomain)
```

Comment:
I'm worried about performance and scaling. We already have a lot of clients. It loads all of them, including all attributes even though only a few are used, and does N+1 domain look up as well.

I'd like to see one or more of the following:

- Add a filter query which supports a list of client ids to load. (For the log  we only need a few clients at a time)
- Only select client attributes which we're going to use.
- Load domains in one query. Can also return an unverified domainName and let the data-loader query the domains.

Reply:
Yes I shared this concern, was going to include the query param filter in second iteration. But that was maybe a bit to much shape up 🤪

I updated this PR to:
* Add the query param to filter by clientIds
* Select only the id and the name attributes
* Return domainName to be queried in the GQL layer.